src/argilla/training/openai.py

```diff
@@ -0,0 +1,225 @@
+        self.model_kwargs["classification_n_classes"] = classification_n_classes
+        self.model_kwargs["classification_positive_class"] = classification_positive_class
+        self.model_kwargs["classification_betas"] = classification_betas
+        self.model_kwargs["suffix"] = suffix
+
+        if isinstance(self._settings, TextClassificationSettings) and self._eval_dataset:
+            label_schema = self._settings.label_schema
+            if len(label_schema) == 2:
+                self.model_kwargs["classification_positive_class"] = label_schema[0]
+                self.model_kwargs["compute_classification_metrics"] = True
+            else:
+                self.model_kwargs["classification_n_classes"] = len(label_schema)
+                self.model_kwargs["compute_classification_metrics"] = True
+
+    def update_config(
+        self,
+        **kwargs,
+    ):
+        """
+        Updates the `model_kwargs` dictionaries with the keyword
+        arguments passed to the `update_config` function.
+        """
+        self.model_kwargs.update(kwargs)
+        self.model_kwargs = filter_allowed_args(self.init_training_args, **self.model_kwargs)
+        keys = []
+        for key, value in self.model_kwargs.items():
+            if value is None:
+                keys.append(key)
+        for key in keys:
+            del self.model_kwargs[key]
+
+    def __repr__(self):
+        formatted_string = []
+        arg_dict = {
+            "'OpenAI.fine_tune'": self.model_kwargs,
+        }
+        for arg_dict_key, arg_dict_single in arg_dict.items():
+            formatted_string.append(arg_dict_key)
+            for key, val in arg_dict_single.items():
+                formatted_string.append(f"{key}: {val}")
+        return "\nhttps://platform.openai.com/docs/api-reference/fine-tune\n" + "\n".join(formatted_string)
+
+    def upload_dataset_to_openai(self, dataset, file_name):
+        import json
+
+        import openai
+
+        for rec in dataset:
+            del rec["id"]
+        with open(file_name, "w") as f:
+            for item in dataset:
+                f.write(json.dumps(item) + "\n")
+        upload_response = openai.File.create(file=open(file_name, "rb"), purpose="fine-tune")
+        self._logger.info(upload_response)
+        file_id = upload_response.id
+        return file_id
+
+    def train(self, output_dir: str = None):
+        """
+        We create a openai.FineTune object from a pretrained model, and send data to finetune it.
+        """
+
+        import openai
+
+        if output_dir is not None:
+            self.model_kwargs["suffix"] = output_dir
+
+        if self._train_dataset is not None and self.model_kwargs["training_file"] is None:
+            self.model_kwargs["training_file"] = self.upload_dataset_to_openai(self._train_dataset, "data_train.jsonl")
+        if self._eval_dataset is not None and self.model_kwargs["validation_file"] is None:
```

Comment:
I think that the `validation_file` is mandatory when `compute_classification_metrics=True`, @davidberenstein1957 can you confirm?

If so, then we should handle this to show a warning and set the `compute_classification_metrics` to `False` in case the `validation_file` is not provided